### PR TITLE
fix(react-components): echarts grab on canvas update cursor and tooltip

### PR DESCRIPTION
**Pull request by hwandersman**
## Overview
* Change cursor to open hand when holding shift while over chart area
* When clicking to pan, change to "grab" hand
* Tooltip is hidden while there is an active gesture

## Verifying Changes

https://github.com/awslabs/iot-app-kit/assets/87385528/f9158635-23f5-45a4-ac22-63a27e261c72

1. Always show the "grab" cursor when shift is held down over the chart canvas
2. Always show the "grabbing" cursor when shift and the mouse is held down over the chart canvas, turn off the tooltip, and drag the canvas left to right
3. Don't show any change in cursor when the mouse is held down
4. Cannot drag on the edges of the canvas, but there is no way to avoid changing the cursor since the event is within the entire canvas

### Scene Composer
For `scene-composer` package changes specifically, you can preview the component in the published storybook artifact. To do this, wait for the `Publish Storybook` action to complete below.

- Click on the workflow details
- Select the Summary item on the left
- Download the zip file

To run the storybook build locally, you need a local static web server:

```
npm install -g httpserver
cd <Extracted Zip Directory>
httpserver
```

Then open the website http://localhost:8080 to run the doc site.

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/react-components/src/components/chart/hooks/useDataZoom.ts [hunk 1]
```diff
@@ -62,4 +62,111 @@ export const useDataZoom = (chartRef: MutableRefObject<EChartsType | null>, view
     // ignoring because refs dont need to be in dependency array
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [setViewport]);
+
+  // When dragging on the chart, turn off tooltip and set cursor to grabbing
+  const setOnDragStart = (chart: EChartsType | null) => {
```
Comment (mnischay) [inline packages/react-components/src/components/chart/hooks/useDataZoom.ts:67]: All these events have nothing to do with DataZoom, handle all the user events in a different file. 
We also have handlers listening to these events in the trendcursors file,  if they are dependent, I would move these there, if not, i would suggest a different file something like handleChartEvents

Comment (mnischay) [reply packages/react-components/src/components/chart/hooks/useDataZoom.ts:67]: Also, all the setoptions happen in one place, so I would return the value using state, i.e the end of the handler would return `{ tooltip: {
        show: false,
      },} ` which then will be used in the last hook which sets the options.

**Review by mnischay — CHANGES_REQUESTED** [review]

File: packages/react-components/src/components/chart/hooks/useDataZoom.ts (hunk 1 above)
Comment (hwandersman) [reply packages/react-components/src/components/chart/hooks/useDataZoom.ts]: Updated! Didn't know `setOptions` is only set in one place, good callout. Moved everything to its own file.

File: packages/react-components/src/components/chart/baseChart.tsx [hunk 2]
```diff
@@ -117,14 +118,18 @@ const BaseChart = ({ viewport, queries, size = { width: 500, height: 500 }, ...o
...
 
+  // handle chart event updates
+  const eventOptions = useHandleChartEvents(chartRef);
+
   // set all the options on the echarts instance
   useEChartOptions(
     chartRef,
     {
       ...convertedOptions,
+      ...eventOptions,
```
Comment (mnischay) [inline packages/react-components/src/components/chart/baseChart.tsx:132]: convertedOptions also has options for tooltip, you probably want to seperate out tooltip options or merge them

File: packages/react-components/src/components/chart/hooks/useHandleChartEvents.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  };
+
+  // Set up event listeners for pressing shift and dragging the mouse to update
+  // the cursor and tooltip
+  useEffect(() => {
+    const chart = chartRef?.current;
+
+    let shiftDown = false;
+    let mouseDown = false;
+    let prevIsGrabbing = false;
```
Comment (mnischay) [inline packages/react-components/src/components/chart/hooks/useHandleChartEvents.ts:35]: for prev value, you probably want to use a useref ?

File: packages/react-components/src/components/chart/hooks/useHandleChartEvents.ts [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    });
+    chart?.getZr().setCursorStyle('grab');
+  };
+
+  // Set up event listeners for pressing shift and dragging the mouse to update
+  // the cursor and tooltip
+  useEffect(() => {
+    const chart = chartRef?.current;
+
+    let shiftDown = false;
```
Comment (mnischay) [inline packages/react-components/src/components/chart/hooks/useHandleChartEvents.ts:33]: curious: is this better than using local state ?

**Review by mnischay — CHANGES_REQUESTED** [review]
The merged of options is the only major call out, rest are minor. Also, please test it along with trendcursors i.e add a couple and see how these event react along with it.

File: packages/react-components/src/components/chart/baseChart.tsx (hunk 2 above)
Comment (hwandersman) [reply packages/react-components/src/components/chart/baseChart.tsx:132]: The convertedOptions tooltip does not specify the `show` parameter. Even if it did we'd want the eventOptions to override it, so this should be safe to do.

File: packages/react-components/src/components/chart/hooks/useHandleChartEvents.ts (hunk 4 above)
Comment (hwandersman) [reply packages/react-components/src/components/chart/hooks/useHandleChartEvents.ts:33]: Not using these variables for any re-renders, so only need to manage the logic within the `useEffect`

File: packages/react-components/src/components/chart/baseChart.tsx (hunk 2 above)
Comment (hwandersman) [reply packages/react-components/src/components/chart/baseChart.tsx:132]: I see `eventOptions` is totally overwriting the `convertedOptions.tooltip`. Will merge the two tooltips.

**Review by mnischay — APPROVED** [review]

**Pull request merged**